docs/guide/ingress/ingress_class.md

```diff
@@ -0,0 +1,140 @@
+
+Before the IngressClass resource and `ingressClassName` field were added in Kubernetes 1.18, Ingress classes were
+specified with a `kubernetes.io/ingress.class` annotation on the Ingress. This annotation was never formally defined,
+but was widely supported by Ingress controllers.
+
+The newer `ingressClassName` field on Ingresses is a replacement for that annotation, but is not a direct equivalent.
+While the annotation was generally used to reference the name of the Ingress controller that should implement the
+Ingress, the field is a reference to an IngressClass resource that contains additional Ingress configuration, including
+the name of the Ingress controller.
+
+!!!tip "disable `kubernetes.io/ingress.class` annotation"
+    In order to maintain backwards-compatibility, `kubernetes.io/ingress.class` annotation is still supported currently.
+    You can enforce IngressClass resource adoption by disable the `kubernetes.io/ingress.class` annotation via [--disable-ingress-class-annotation](../../../deploy/configurations/#disable-ingress-class-annotation) controller flag.
+
+## IngressClassParams
+IngressClassParams is a CRD specific to AWS Load Balancer Controller, which can be used along with IngressClass’s parameter field.
+You can use IngressClassParams to enforce settings for a set of Ingresses.
+
+!!!example
+    - with scheme & ipAddressType & tags
+    ```
+    apiVersion: elbv2.k8s.aws/v1beta1
+    kind: IngressClassParams
+    metadata:
+      name: awesome-class
+    spec:
+      scheme: internal
+      ipAddressType: dualstack
+      tags:
+      - key: org
+        value: my-org
+    ```
+    - with namespaceSelector
+    ```
+    apiVersion: elbv2.k8s.aws/v1beta1
+    kind: IngressClassParams
+    metadata:
+      name: awesome-class
+    spec:
+      namespaceSelector:
+        matchLabels:
+          team: team-a
+    ```
+    - with IngressGroup
+    ```
+    apiVersion: elbv2.k8s.aws/v1beta1
+    kind: IngressClassParams
+    metadata:
+      name: awesome-class
+    spec:
+      group:
+        name: my-group
+    ```
+
+### IngressClassParams specification
+
+#### spec.namespaceSelector
+`namespaceSelector` is an optional setting, It follows standard label selector semantics.
+
+Cluster administrators can use `namespaceSelector` field to restrict the namespaces of Ingresses that are allowed to specify the IngressClass.
+
+1. If `namespaceSelector` specified, only Ingresses in selected namespaces can use IngressClasses with this parameter. The controller will refuse to reconcile for Ingresses that violates `namespaceSelector`.
+2. If `namespaceSelector` un-specified, all Ingresses in any namespace can use IngressClasses with this parameter.
+
+#### spec.group
+
+`group` is an optional setting.  The only available sub-field is `group.name`.
+
+Cluster administrators can use `group.name` field to denote the groupName for all Ingresses belong to this IngressClass.
```

Comment:
```suggestion
Cluster administrators can use `group.name` field to denote the ALB ingress group for all Ingresses belong to this IngressClass. (IngressGroup is a CustomResourceDefinition specific to the AWS Load Balancer Controller)
```
?

Reply:
This is inaccurate. IngressGroup is not a CRD. it's sort of virtual grouping for Ingresses with same group name.

Reply:
OK - then I recommend rewording to introduce this controller's ingress group concept and explain it's _not_ a resource.